[IMP] accounting: info about proforma label on invoices (#11080)
odoo/documentation: content/applications/finance/accounting/customer_invoices.rst

fees, and other charges.
Odoo supports multiple invoicing and payment workflows.

.. seealso::
   - :doc:`/applications/finance/accounting/customer_invoices/overview`

From draft invoice to profit and loss report, the process involves several steps once the goods are
shipped to a customer:

- :ref:`accounting/invoice/creation`
- :ref:`accounting/invoice/confirmation`
- :ref:`accounting/invoice/sending`
- :ref:`accounting/invoice/paymentandreconciliation`
- :ref:`accounting/invoice/followup`
- :ref:`accounting/invoice/reporting`

.. note::
   Odoo :guilabel:`Invoicing` is a standalone app designed to create invoices, send them to
   customers, and manage payments. On the other hand, the :guilabel:`Accounting` app is a
   comprehensive accounting solution that allows the same actions to be performed.

.. _accounting/invoice/creation:

Invoice creation
================

Draft invoices can be manually created from documents like sales orders or purchase orders or
directly from the :guilabel:`Customer Invoices` journal in the :guilabel:`Accounting Dashboard`.

An invoice must include the required information to enable the customer to pay promptly for the

> goods and services ordered and delivered. Make sure the following fields are appropriately

Author: Yes, you're right! :) Would it be ok just to change "... to pay promptly for the goods and services ordered and delivered" to "... to pay promptly for the goods and services ordered/delivered"? Thanks ;)

> goods and services ordered and delivered. Make sure the following fields are appropriately

Reviewer: What if we simplify it even further and just say "... to pay promptly for their goods and services." ?

completed:

- :guilabel:`Customer`: Odoo automatically fills in some information based on the customer’s
  registered information and previous invoices.
- :guilabel:`Invoice Date`: is automatically filled in when the invoice is confirmed.
- :guilabel:`Payment Reference`: is the reference to be set as the payment reference to facilitate

>   reconciliation. It's automatically filled in when the invoice is confirmed.

Author: Ok, thanks ;)

- :guilabel:`Due Date` or :guilabel:`Payment terms`: to specify when the customer has to pay the
  invoice.
- :guilabel:`Journal`: is automatically set and can be changed if needed.
- :doc:`Currency <get_started/multi_currency>`
- :guilabel:`Product`
- :guilabel:`Quantity`
- :guilabel:`Price`
- :guilabel:`Taxes` (if applicable).

To display the total amount of the invoice in words, go to :menuselection:`Accounting -->
Configuration --> Settings` and activate the :guilabel:`Total amount of invoice in letters` option.

The :guilabel:`Journal Items` tab displays the accounting entries created.
To add specific invoice/accounting information, fill in the :guilabel:`Other Info` tab.

The system initially creates invoices in :guilabel:`Draft` status. While they remain unvalidated,
draft invoices have no accounting impact within the system. When previewing a draft invoice,

> :guilabel:`Proforma` is added before the invoice reference, as proforma invoices are issued to

Author: In 17, it's still possible to preview the draft invoice (with the pro-forma mention on it), but it's not possible to send it to the customer unless the option is enabled in the Sales app. I probably need to be more specific then as I added this information following a request in task 3938154 (The customer is concerned that the word "PROFORMA" will appear when previewing the invoice. While this is standard and improved from version 17.0+e, the documentation does not reflect this update)

> :guilabel:`Proforma` is added before the invoice reference, as proforma invoices are issued to

Author: In 17, it's still possible to preview the draft invoice (with "proforma" mentioned on it). The customer can download it from the customer portal, but it's not possible to send it to him. If the "pro-forma" option is enabled in the Sales app, it's possible to "print without amount" (but an error occurs every time I try it). I probably need to be more specific then, as I added this information following a request in task 3938154 (The customer is concerned that the word "PROFORMA" will appear when previewing the invoice. While this is standard and improved from version 17.0+e, the documentation does not reflect this update)

request payment from the buyer before the goods or services are provided.

.. _accounting/invoice/confirmation:

Invoice confirmation
====================

Click :guilabel:`Confirm` when the document is completed. The document's status changes to
:guilabel:`Posted`, and a journal entry is generated based on the invoice configuration. The

> :guilabel:`Journal Items` tab is then filled.

Author: Yes, you're right, I got them mixed up (Journal item tab and Journal items in the Accounting menu) :)
I can remove the image as well.

.. image:: customer_invoices/invoice-confirmation.png
   :alt: invoice confirmation journal entries

On confirmation, Odoo assigns each document a unique number from a defined sequence consisting of a
prefix and a number. The prefix combines the journal code and the entry date. It is used to classify
entries by period. The number is unique for each period and is used to identify the entry. The
default sequence on customer invoices is INV/YYYY/number. In some specific cases,
:ref:`resequencing <accounting/invoice/resequencing>` might be necessary.

When previewing a confirmed invoice, :guilabel:`Proforma` is added before the invoice reference, as
proforma invoices are issued to request payment from the buyer before the goods or services are
provided.

.. note::
   Once confirmed, an invoice can no longer be updated. Click :guilabel:`Reset to draft` if
   changes are required.

.. _accounting/invoice/sending:

Invoice sending
===============

> To send the invoice to the customer, click :guilabel:`Send & Print`. A :guilabel:`Configure your

Author: Yes, sure, thanks for the reminder ;)

document layout` pop-up window will appear if a :ref:`default invoice layout
<studio/pdf-reports/default-layout>` hasn't been customized.
To send it to the customer's email address and download a PDF of the document, click

> :guilabel:`Send & Print` (enable the :guilabel:`By Post` option if needed).

Author: Ok ;) Maybe I shouldn't keep the image then?

- To exclude any invoices from the follow-up, enable the option :guilabel:`Exclude from Follow-ups`
  for the selected ones.

.. _accounting/invoices/aging-report:

Aged Receivable
---------------

To review outstanding customer invoices and their related due dates, use the
:ref:`Aged Receivable <accounting/reporting/aged-receivable>` report. To access it, go to
:menuselection:`Accounting --> Reporting --> Aged Receivable`.

.. _accounting/invoice/reporting:

Reporting
=========

Profit and Loss
---------------

The :ref:`Profit and Loss <accounting/reporting/profit-and-loss>` statement shows details of income
and expenses.

Balance sheet
-------------

The :ref:`Balance Sheet <accounting/reporting/balance-sheet>` summarizes the company's liabilities,
assets, and equity at a specific time.

.. _accounting/invoice/resequencing:

Resequencing
============

:guilabel:`Resequence` is a technical feature used to change the sequence of documents. It can be
helpful to resequence invoice numbers when importing invoices from another invoicing or accounting
system, mainly if the reference originates from the previous software and continuity for the current

> year must be maintained without restarting from the beginning.

Reviewer: It is also possible to change the sequence of invoices to include information such as the month and year. I actually learned something new from this forum post! https://www.odoo.com/forum/help-1/how-to-resequence-invoices-in-odoo-with-month-and-year-257878#:~:text=If%20we%20want%20to%20add,term%20as%20year%20and%20month.
I'm not sure if we want to include this as another method of resequencing or try to separate it a bit. Let's discuss when we're both back :)

Reviewer: Maybe this section should be called Sequences and then we can have a subsection that is "Resequencing". The first would explain what they are, how to set them as explained by that forum post, etc. The subsection would explain how to resequence them as you have explained.

Author: Should I take the paragraph from the 'Invoice confirmation' part for the new 'Sequence' section? I think what the forum post explains should be in the 'Resequencing' section as they change the sequence.

Reviewer: I was thinking of how new users might want to set their invoices up like this from the beginning, but you're right that it is technically resequencing. Let's call what you've described "mass resequencing" and the method of the forum post "resequencing". This will bring a few changes:
- Mass resequencing is only available to users with administrator or advisor Accounting access (not the case for individual resequencing).
- Neither requires developer mode as of my testing; let me know if you find otherwise.

The sequence can be modified once an invoice is in the system.

> .. note::

Reviewer: Great note. I would also add that sequence changes also affect the format of future invoices' sequence.

   - This feature is only available to users with administrator or advisor access.
   - All sequence changes are logged in the chatter to keep the information.

> Here are the different steps to resequence invoice numbers:

Reviewer: Instead of "here are the different steps" let's try "Follow these steps"

#. Activate the :ref:`developer mode <developer-mode>`.
#. From the :guilabel:`Accounting Dashboard`, open the :guilabel:`Customer Invoices` journal.
#. Select the documents that need a new sequence.
#. In the :icon:`fa-cog` :guilabel:`Actions` menu, click :guilabel:`Resequence`.
#. In the :guilabel:`Ordering` field, choose to

   - :guilabel:`Keep current order`: the order of the numbers remains the same
   - :guilabel:`Reorder by accounting date`: the number is reordered by accounting date.
#. Set the :guilabel:`First New Sequence`.
#. :guilabel:`Preview Modifications` and click :guilabel:`Confirm`.

.. image:: customer_invoices/invoice-sequencing.png
   :alt: Resequence options window

In some cases, resequencing is not possible:

> - When entries are previous to a lock date.

Reviewer: ```suggestion
- When entries are before a lock date.
```

> - When changing the date of a previously posted invoice from another period, the sequence is not

Reviewer: Can we get an example for this one so it's more clear?

Author: I also changed the text to make it more clear.

Customer invoices
=================

A customer invoice is a document issued by a company for products and/or services sold to a
customer. It records receivables as they are sent to customers. Customer invoices can include
amounts due for the goods and/or services provided, applicable sales taxes, shipping and handling
fees, and other charges.
Odoo supports multiple invoicing and payment workflows.

.. seealso::
   - :doc:`/applications/finance/accounting/customer_invoices/overview`

From draft invoice to profit and loss report, the process involves several steps once the goods or

> services are ordered or shipped to a customer, depending on the invoicing policy:

Author: @dade-odoo Thanks for explaining the nuances of using this tense ;)

- :ref:`accounting/invoice/creation`
- :ref:`accounting/invoice/confirmation`
- :ref:`accounting/invoice/sending`
- :ref:`accounting/invoice/paymentandreconciliation`
- :ref:`accounting/invoice/followup`
- :ref:`accounting/invoice/reporting`

.. note::
   Odoo :guilabel:`Invoicing` is a standalone app designed to create invoices, send them to
   customers, and manage payments. It also handles flows involving vendor bills. On the other hand,
   the :guilabel:`Accounting` app is a comprehensive accounting solution that allows the same
   actions and includes additional features such as standard financial reports, bank reconciliation,

>    budget, asset management, and more.

Author: @dade-odoo I didn't think it could have 2 meanings with just one "s"! Thanks for the explanation ;) (I choose the second option)

   - If required, invoices and other journal entries can be locked once posted
     using the :ref:`Lock posted entries with hash <data-inalterability/lock>` feature.

.. _accounting/invoice/sending:

Invoice sending
===============

To send the invoice to the customer, click :guilabel:`Send & Print`. A :guilabel:`Configure your
document layout` pop-up window will appear if a :ref:`default invoice layout
<studio/pdf-reports/default-layout>` hasn't been customized. Then, select the ways to send this
invoice to the customer in the :guilabel:`Send` window.

To send and print multiple invoices, go to the list view and select them. In the :icon:`fa-cog`
:guilabel:`Actions` menu, select the :guilabel:`Send & Print` option. A banner will appear on the
selected invoices to indicate they are part of an ongoing send and print batch. This helps prevent
the process from being triggered manually again, as it may take some time to complete for
exceptionally large batches.

.. _accounting/invoice/paymentandreconciliation:

Payment and reconciliation
==========================

In Odoo, an invoice is considered :guilabel:`Paid` when the associated accounting entry has been

> reconciled with the payment entries.

Author: @dade-odoo You're right, thanks for the reminder ;)

.. seealso::
   - :doc:`payments`
   - :doc:`bank/reconciliation`

.. _accounting/invoice/followup:

Payment follow-up
=================

Odoo helps define a :doc:`follow-up <payments/follow_up>` strategy. Different actions can be set up
to remind customers to pay their outstanding invoices, depending on how much the customer is
overdue. These actions are bundled into follow-up levels that trigger when an invoice is overdue by
a certain number of days. If there are other overdue invoices for the same customer, the actions are
performed on the most overdue invoice.

.. _accounting/invoice/followup-reports:

> Follow-up Reports

Reviewer: This section seems like it repeats a lot of information that's already covered on the follow-ups page itself. I'll leave it for now, but let's ask Audrey what she thinks.

Author: @dade-odoo You're right, I'll get rid of this section and just keep the explanation above where I referenced the "Follow-up on invoices" page. Thanks!

-----------------

To set the different follow-up levels, go to
:menuselection:`Accounting --> Configuration --> Follow-up Levels`. All actions triggered after a
specified number of days can be viewed. To manage the various options, such as the number of days,
the type of reminder sent, or whether the action is automated, click on the specific line that needs
to be updated.

Then, follow these steps to take the appropriate action:

- To access the list of customers requiring action along with the :guilabel:`Total Due` and
  :guilabel:`Total Overdue` amounts, go to
  :menuselection:`Accounting --> Customers --> Follow-up Reports`.
- To get a full overview of the customer's open invoices, click on its line.
- To open a payment reminder to be sent to the customer, click the :guilabel:`Follow up` button.
- To exclude any invoices from the follow-up, enable the option :guilabel:`Exclude from Follow-ups`
  for the selected ones.

.. _accounting/invoices/aging-report:

> Aged Receivable

Reviewer: I understand why this could belong under Payment Follow-ups, but I think it would fit better in the Reporting section and could be paired with the Aged Payable and Partner Ledger. Let's have a "Partner Reports" section under "Reporting" that explains the three reports briefly. It could be as short as one line per report like the P&L and BS have or a little longer if you have more to say (since they don't have their own pages like the P&L and BS).

---------------

To review outstanding customer invoices and their related due dates, use the
:ref:`Aged Receivable <accounting/reporting/aged-receivable>` report. To access it, go to
:menuselection:`Accounting --> Reporting --> Aged Receivable`.

.. _accounting/invoice/reporting:

Reporting
=========

Profit and Loss
---------------

The :ref:`Profit and Loss <accounting/reporting/profit-and-loss>` statement shows details of income
and expenses.

Balance sheet
-------------

> The :ref:`Balance Sheet <accounting/reporting/balance-sheet>` summarizes the company's liabilities,
> assets, and equity at a specific time.

Reviewer: ```suggestion
The :ref:`Balance Sheet <accounting/reporting/balance-sheet>` summarizes the company's assets,
liabilities, and equity at a specific time.
```
I'm not sure why but it sounds odd to me to have "liabilities" first - maybe because they're often taught with the acronym ALE to remember Assets = Liabilities + Equity, so let's use that order :)

.. _accounting/invoice/sequence:

Sequence
========

The sequence that Odoo assigns to each document is a unique number made up of a prefix and a number.

> The prefix combines the journal code and the entry date. It is used to classify entries by period.

Reviewer: ```suggestion
The prefix combines the journal code and the entry date and is used to group entries by period.
```
Feels more natural than the two short sentences with the same subject :)
"Classify" is technically correct but sounds kind of technical and we want to make it sound simple :)

amounts due for the goods and/or services provided, applicable sales taxes, shipping and handling
fees, and other charges.
Odoo supports multiple invoicing and payment workflows.

.. seealso::
   - :doc:`/applications/finance/accounting/customer_invoices/overview`

From draft invoice to profit and loss report, the process involves several steps once the goods (or
services) have been ordered/shipped (or rendered) to a customer, depending on the invoicing policy:

- :ref:`accounting/invoice/creation`
- :ref:`accounting/invoice/confirmation`
- :ref:`accounting/invoice/sending`
- :ref:`accounting/invoice/paymentandreconciliation`
- :ref:`accounting/invoice/followup`
- :ref:`accounting/invoice/reporting`

.. note::
   Odoo :guilabel:`Invoicing` is a standalone app designed to create invoices, send them to
   customers, and manage payments. It also handles flows involving vendor bills. On the other hand,
   the :guilabel:`Accounting` app is a comprehensive accounting solution that allows the same
   actions and includes additional features such as standard financial reports, bank reconciliation,

>    budgets, asset management, and more.

Reviewer 2: You could also change that one note on the Accounting and Invoicing page in this PR and change the rest of that page later if you want :wink:

manually from the :guilabel:`Customer Invoices` journal in the :guilabel:`Accounting Dashboard`.

An invoice must include the required information to enable the customer to pay promptly for their
goods and services. Make sure the following fields are appropriately completed:

- :guilabel:`Customer`: When a customer is selected, Odoo automatically pulls information from the
  customer record like the invoice address,
  :doc:`preferred payment terms <customer_invoices/payment_terms>`,
  :doc:`fiscal positions <taxes/fiscal_positions>`, receivable account, and more onto the invoice.
  To change these values for this specific invoice, edit them directly on the invoice. To change
  them for future invoices, change the values on the contact record.
- :guilabel:`Invoice Date`: If not set manually, this field is automatically set as the current date
  upon confirmation.
- :guilabel:`Due Date` or :doc:`payment terms <customer_invoices/payment_terms>`: To specify when
  the customer has to pay the invoice.
- :guilabel:`Journal`: Is automatically set and can be changed if needed.
- :doc:`Currency <get_started/multi_currency>`

> - :guilabel:`Product`

Author: @auva-odoo yes, sure, it's worth mentioning!

selected invoices to indicate they are part of an ongoing send and print batch. This helps prevent
the process from being triggered manually again, as it may take some time to complete for
exceptionally large batches.

.. _accounting/invoice/paymentandreconciliation:

Payment and reconciliation
==========================

In Odoo, an invoice is considered :guilabel:`Paid` when the associated accounting entry has been
reconciled with a corresponding bank transaction.

.. seealso::
   - :doc:`payments`
   - :doc:`bank/reconciliation`

.. _accounting/invoice/followup:

Payment follow-up
=================

Odoo's :doc:`follow-up actions <payments/follow_up>` help companies follow up on customer invoices.
Different actions can be set up to remind customers to pay their outstanding invoices, depending on
how much the customer is overdue. These actions are bundled into follow-up levels that trigger when
an invoice is overdue by a certain number of days. If there are multiple overdue invoices for the
same customer, the actions are performed on the most overdue invoice.

.. _accounting/invoice/reporting:

> Reporting

Author: @auva, yes, it makes sense ;) Thanks for the advice.
